### DISCUSSION TITLE
Tus support

### DISCUSSION THREAD

**Discussion by WagnerPMC** [roadrunner-server/roadrunner#1032] [Q&A] [answered]
Hello there!

Who knows if RoadRunner supports the Tus protocol (tus.io)?
I mean specifically a solution that works with RR + tus-php (https://github.com/ankitpokhrel/tus-php)

**Comment by rustatian** [comment, marked as answer]
@GavnoMocha Hello. RR doesn't support the Tus protocol, but it can be implemented as an `http` plugin middleware (handle a `tus` headers).

**Reply by WagnerPMC** [reply]
Thank you

**Reply by ChristianPavilonis** [reply]
@GavnoMocha Did you ever get tus-php working with roadrunner?

**Reply by WagnerPMC** [reply]
Yes, I have working sketches. But I haven't figured out how to pass the names of the uploaded files in php

**Reply by WagnerPMC** [reply]
I saw you recently added streaming capabilities, I think we can implement a small blood through php, but this is not quite right, because we incapacitate the employee with this download

**Reply by WagnerPMC** [reply]
Therefore, I think it is more correct to do it through RoadRunner
